WorkloadManagementSystem/Utilities/QueueUtilities.py

```diff
@@ -0,0 +1,225 @@
+        queueDict[queueName] = qDict[queue]
+        queueDict[queueName]['Queue'] = queue
+        queueDict[queueName]['Site'] = site
+        # Evaluate the CPU limit of the queue according to the Glue convention
+        # To Do: should be a utility
+        if "maxCPUTime" in queueDict[queueName] and \
+           "SI00" in queueDict[queueName]:
+          maxCPUTime = float(queueDict[queueName]['maxCPUTime'])
+          # For some sites there are crazy values in the CS
+          maxCPUTime = max(maxCPUTime, 0)
+          maxCPUTime = min(maxCPUTime, 86400 * 12.5)
+          si00 = float(queueDict[queueName]['SI00'])
+          queueCPUTime = 60. / 250. * maxCPUTime * si00
+          queueDict[queueName]['CPUTime'] = int(queueCPUTime)
+
+        # Tags & RequiredTags defined on the Queue level and on the CE level are concatenated
+        # This also converts them from a string to a list if required.
+        for tagFieldName in ('Tag', 'RequiredTag'):
+          ceTags = ceDict.get(tagFieldName, [])
+          if isinstance(ceTags, basestring):
+            ceTags = fromChar(ceTags)
+          queueTags = queueDict[queueName].get(tagFieldName)
+          if queueTags and isinstance(queueTags, basestring):
+            queueTags = fromChar(queueTags)
+            queueDict[queueName][tagFieldName] = queueTags
+          if ceTags:
+            if queueTags:
+              allTags = list(set(ceTags + queueTags))
+              queueDict[queueName][tagFieldName] = allTags
+            else:
+              queueDict[queueName][tagFieldName] = ceTags
+
+        # Some parameters can be defined on the CE level and are inherited by all Queues
+        for parameter in ['MaxRAM', 'NumberOfProcessors', 'WholeNode']:
+          queueParameter = queueDict[queueName].get(parameter)
+          ceParameter = ceDict.get(parameter)
+          if ceParameter or queueParameter:
+            queueDict[queueName][parameter] = ceParameter if not queueParameter \
+                else queueParameter
+
+        # If we have a multi-core queue add MultiProcessor tag
+        if queueDict[queueName].get('NumberOfProcessors', 1) > 1:
+          queueDict[queueName].setdefault('Tag', []).append('MultiProcessor')
+
+        queueDict[queueName]['CEName'] = ce
+        queueDict[queueName]['GridCE'] = ce
+        queueDict[queueName]['CEType'] = ceDict['CEType']
+        queueDict[queueName]['GridMiddleware'] = ceDict['CEType']
+        queueDict[queueName]['QueueName'] = queue
+
+        platform = ''
+        if "Platform" in queueDict[queueName]:
+          platform = queueDict[queueName]['Platform']
+        elif "Platform" in ceDict:
+          platform = ceDict['Platform']
+        elif "OS" in ceDict:
+          architecture = ceDict.get('architecture', 'x86_64')
+          platform = '_'.join([architecture, ceDict['OS']])
+
+        queueDict[queueName]['Platform'] = platform
+        if "Platform" not in queueDict[queueName] and platform:
```

Comment:
The first half of this condition will never be true as it's set on the line above